### PR TITLE
Change Parse.Promise to be a class

### PR DESCRIPTION
**Pull request by egistli**
As [Parse JS SDK documentation](https://parse.com/docs/js/api/classes/Parse.Promise.html), one should be able to create `Promise` objects with `Parse.Promise.error()`, `Parse.Promise.as()`. (or `new` keyword)

This pull request change the type of `Promise` from `interface` to `class` and also update `is`, `error`, `as` and `when` methods to be static method with corresponding arguments.

With this change, we can utilize `Promise` functionality provided by Parse.

### CODE REVIEW

**Comment by egistli** [comment]
One thing I am not sure is do we need the `IPromise` interface?
Or we want it to be a fully Promise compatible interface for future implementation, maybe we should add methods like `reject`, `done`, `resolve`... etc onto it.

**Comment by vvakame** [comment]
@ejferg could you review this PR?

File: parse/parse.d.ts [hunk 1]
```diff
@@ -72,14 +72,16 @@ declare namespace Parse {
         then<U>(resolvedCallback: (value: T) => U, rejectedCallback?: (reason: any) => U): IPromise<U>;
     }
 
-    interface Promise<T> {
+    class Promise<T> {
+
+        static as(resolvedValue: any): Promise<any>;
+        static error(error: any): Promise<any>;
```
Comment (DanielRosenwasser) [inline parse/parse.d.ts]: I would think that these should take a type parameter `T` and return a `Promise<T>`

Comment (egistli) [inline parse/parse.d.ts]: Ah, you are right. I'll fix that.

**Comment by vvakame** [comment]
oops!
thanks mate!

**Pull request merged**